nixos/modules/services/misc/autosuspend.nix

```diff
@@ -0,0 +1,238 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.services.autosuspend;
+
+  settingsFormat = pkgs.formats.ini {};
+
+  checks =
+    mapAttrs'
+    (n: v: {
+      name = "check.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.checks;
+  wakeups =
+    mapAttrs'
+    (n: v: {
+      name = "wakeup.${n}";
+      value = filterAttrs (_: v: v != null) v;
+    })
+    cfg.wakeups;
+
+  # Whether the `Smb` check is enabled, and therefore requires `smbstatus` in path
+  needsSamba =
+    (filterAttrs (n: v:
+      (
+        if v.class == null
+        then n
+        else v.class
+      )
+      == "Smb"
+      && v.enabled)
+    cfg.checks)
+    != {};
+
+  autosuspend-conf =
+    settingsFormat.generate "autosuspend.conf" ({general = cfg.settings;} // checks // wakeups);
+
+  autosuspend = cfg.package;
+
+  version = "4.3.0";
+
+  checkType = with types;
+    submodule {
+      freeformType = settingsFormat.type.nestedTypes.elemType;
+
+      options.enabled = mkEnableOption (mdDoc "this activity check");
```

Comment:
I agree that it would make sense.  I default to `false` because that's what `autosuspend` does in its config: you need to explicitly write `enabled = true` to enable a check.  I'm not sure whether I should change that default :thinking:

Reply:
Yep that's fair.  I'm gonna go ahead and flip the default, I agree it makes the whole interface friendlier.